cli/src/test/java/uk/ac/ebi/atlas/cli/experiment/CreateUpdateExperimentCommandTest.java

```diff
@@ -0,0 +1,129 @@
+package uk.ac.ebi.atlas.cli.experiment;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import picocli.CommandLine;
+import uk.ac.ebi.atlas.experimentimport.ExperimentChecker;
+import uk.ac.ebi.atlas.experimentimport.ExperimentCrudDao;
+import uk.ac.ebi.atlas.experimentimport.GxaExperimentCrud;
+import uk.ac.ebi.atlas.experimentimport.condensedSdrf.CondensedSdrfParser;
+import uk.ac.ebi.atlas.experimentimport.experimentdesign.ExperimentDesignFileWriterService;
+import uk.ac.ebi.atlas.experimentimport.idf.IdfParser;
+import uk.ac.ebi.atlas.trader.ConfigurationTrader;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CreateUpdateExperimentCommandTest {
+
+    private CreateUpdateExperimentCommand createUpdateExperimentCommand;
+
+    private GxaExperimentCrud gxaExperimentCrudSpy;
+
+    @Mock
+    private ExperimentCrudDao experimentCrudDao;
+    @Mock
+    private ExperimentDesignFileWriterService experimentDesignFileWriterService;
+    @Mock
+    private ConfigurationTrader configurationTrader;
+    @Mock
+    private ExperimentChecker experimentChecker;
+    @Mock
+    private CondensedSdrfParser condensedSdrfParser;
+    @Mock
+    private IdfParser idfParser;
+
+    private List<UUID> randomUUIDs;
+
+    @Before
+    public void setUp() {
+        gxaExperimentCrudSpy = new GxaExperimentCrud(experimentCrudDao, experimentDesignFileWriterService,
+                configurationTrader, experimentChecker, condensedSdrfParser, idfParser);
+
+        randomUUIDs = generateRandomUUIDs(10);
+    }
+
+    @Test
+    public void whenNoAccessionPassed_ThenReturnCorrectExitCode() {
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(gxaExperimentCrudSpy);
+        var exitCode = createUpdateExperimentCommand.call();
+
+        final int exitCodeForNoExperimentAccessionSupplied = 1;
+        assertThat(exitCode).isEqualTo(exitCodeForNoExperimentAccessionSupplied);
+    }
+
+    @Test
+    public void createExperimentHasBeenCalledByNumberOfProvidedAccessions() {
+        GxaExperimentCrud spiedGxaExperimentCrud = spy(gxaExperimentCrudSpy);
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(spiedGxaExperimentCrud);
+        final UUID uuid1 = randomUUIDs.get(0);
+        final UUID uuid2 = randomUUIDs.get(1);
+        final UUID uuid3 = randomUUIDs.get(2);
```

Comment:
Do you need these? I think you can inline `randomUUIDs.get(...)` in the `doReturn` below.

Reply:
agreed, done